src/ui/hexedit.cc

```diff
@@ -108,7 +108,8 @@ HexEdit::HexEdit(FileBlobModel *dataModel, QItemSelectionModel *selectionModel,
       selection_size_(0),
       current_area_(WindowArea::HEX),
       cursor_pos_in_byte_(0),
-      cursor_visible_(false) {
+      cursor_visible_(false),
+      edit_engine_(dataModel->binData().width()) {
```

Comment:
on new bin data, when width is diffrent you should probbably recreate edit_engine. Think about situation when there is button in UI: change byte width.